File: server/java/src/main/java/com/onlinepayments/example/service/interfaces/handlers/PaymentMethodHandlerInterface.java
package com.onlinepayments.example.service.interfaces.handlers;

import com.onlinepayments.example.service.domain.payments.enums.PaymentMethodType;
import com.onlinepayments.example.service.dtos.payments.CreatePaymentRequestDto;
import com.onlinepayments.example.service.dtos.payments.CreatePaymentResponseDto;

public interface PaymentMethodHandlerInterface {
    boolean isPaymentMethodSupported(PaymentMethodType method);

    CreatePaymentResponseDto createPayment(CreatePaymentRequestDto request);
}

File: server/nodejs/src/business/interfaces/payment-method-handler-interface.ts
import { PaymentMethodType } from '../domain/enums/payments/payment-method-type';
import { CreatePaymentRequestDto } from '../dtos/payment/create-payment/create-payment-request-dto';
import { CreatePaymentResponseDto } from '../dtos/payment/create-payment/create-payment-response-dto';

export interface PaymentMethodHandlerInterface {
    getSupportedMethod(): PaymentMethodType;
    handle(requestDto: CreatePaymentRequestDto): Promise<CreatePaymentResponseDto>;
}
